scripts/installation/install-openkat.yml

```diff
@@ -0,0 +1,531 @@
+#        authorized_keys. After that run the following command (and don't forget
+#        the comma!
+#
+#        ansible-playbook -vvv install-openkat.yml -i <hostname or ip>, -u root
+
+---
+
+- name: Playbook to install OpenKAT 1.10.1 on Debian 12
+  hosts: all
+  become: true
+  become_user: root
+
+  vars:
+    katrelease: "1.11.0"
+    xtdbrelease: "1.0.7"
+    arch: "amd64"
+
+  tasks:
+
+  - name: First let's check if this is indeed a Debian 12 server.
+    fail:
+      msg: "This playbook only works for a Debian 12 server. But this is a {{ ansible_facts['distribution'] ansible_facts['distribution_major_version'] }} server."
+    when: not ( ansible_facts['distribution'] == "Debian" and ansible_facts['distribution_major_version'] == "12" )
+
+  - name: Generate the random passwords we need during this play.
+    ansible.builtin.set_fact:
+      # Generate unique passwords for several services.
+      # Turns out that special characters create problems!
+      #var: "{{ lookup('community.general.random_string', length=20, min_numeric=3, min_upper=3, min_special=3) }}"
+      rocky_db_password: "{{ lookup('password', '/dev/null', length=20, chars=['ascii_letters'] ) }}"
+      # Generate a unique secure password for the KAT-alogus database user. You can use the same method we used for generating the Rocky database user password.
+      kat_password: "{{ lookup('password', '/dev/null', length=20, chars=['ascii_letters'] ) }}"
+      bytes_password: "{{ lookup('password', '/dev/null', length=20, chars=['ascii_letters'] ) }}"
+      mula_password: "{{ lookup('password', '/dev/null', length=20, chars=['ascii_letters'] ) }}"
+      # The next password needs minimum 2 uppercase, 2 lowercase and 2 special characties.
+      # Because community.general.random_string is only available in recent versions of Ansible we fix these mandatory syntax rules this way.
+      admin_password: "{{ lookup('password', '/dev/null', length=20, chars=['ascii_letters'] ) }}AaBb12-_!"
+      # DJANGO_CSRF_TRUSTED_ORIGINS="https://openkat.example.org"
+      # TODO read IPv4 and IPv6 dict of IP's and add them all here separated with a comma.
+      django_csrf: "https://{{ ansible_default_ipv4.address }}"
+      # Generate a safe password for the KAT user in rabbitmq. You can use the /dev/urandom method again and put it in a shell variable to use it later:
+      # rabbitmq_pass=$(tr -dc A-Za-z0-9 < /dev/urandom | head -c 20). Well, you can, but this works too. :-)
+      kat_mq_password: "{{ lookup('password', '/dev/null length=20 chars=ascii_letters') }}"
+
+  # If you will be running the database on the same machine as KAT, you can install Postgres:
+  # sudo apt install postgresql
+  - name: Install needed packages
+    ansible.builtin.apt:
+      name: "{{ item }}"
+      state: present
+    loop:
+      - python3-psycopg2
+      - python3-pexpect
+      - postgresql
+      - postgresql-client
+      - rabbitmq-server
+      - sudo
```

Comment:
Dit is heel traag en on-efficient.... vervang de loop door: pkg:
Dan hoeft de apt-taak maar 1x te draaien

apt:
  pkg:
    - python3-pexpect
    - postgresql 
    ...
  state: present